File: support-files/src/pipeline-plugin/bksdk/src/main/java/com/tencent/bk/devops/atom/pojo/Result.java
package com.tencent.bk.devops.atom.pojo;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Result<T> {
    private int status; //状态码，0代表成功
    private String message; //描述信息
    private T  data; //数据对象

    public Result() {
    }

    public Result(T data) {
        this.status = 0;
        this.data = data;
    }

    public Result(int status, String message) {
        this.status = status;
        this.message = message;
        this.data = null;
    }

}

File: src/backend/turbo/common-turbo/common-turbo-api/src/main/kotlin/com/tencent/devops/common/api/pojo/Result.kt
package com.tencent.devops.common.api.pojo

import com.fasterxml.jackson.annotation.JsonIgnore
import io.swagger.annotations.Api
import io.swagger.annotations.ApiModelProperty

/**
 *
 * Powered By Tencent
 */
@Api("数据返回包装模型")
data class Result<out T>(
    @ApiModelProperty("状态码", required = true)
    val status: Int,
    @ApiModelProperty("错误信息", required = false)
    val message: String? = null,
    @ApiModelProperty("数据", required = false)
    val data: T? = null
) {
    constructor(data: T) : this(0, null, data)
    constructor(status: Int, message: String) : this(status, message, null)

    @JsonIgnore
    fun isOk(): Boolean {
        return status == 0
    }

    @JsonIgnore
    fun isNotOk(): Boolean {
        return status != 0
    }
}
